apps/web/src/pages/[document].tsx

```diff
@@ -55,22 +57,26 @@ export const getServerSideProps: GetServerSideProps<Props> = async (ctx) => {
 const DocumentPage: NextPage<Props> = ({ page: initialPage }) => {
   const [page, setPage] = useState<DeserializedPage | null>(null);
   const [isEditing, setIsEditing] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
 
   const { address } = useAccount();
 
   const handlePageLoad = useCallback(async () => {
+    setIsLoading(true);
```

Comment:
I guess this could have been a `react-query` query?

Reply:
Added now with optimistic updates 🚀